deso/block.go

```diff
@@ -271,6 +271,32 @@ func (node *Node) GetTransactionsForConvertBlock(block *lib.MsgDeSoBlock) []*typ
 		transactions = append(transactions, transaction)
 	}
 
+	// Create a dummy transaction for the "block level" operations
+	// when we have the additional slice of utxo operations.
+	if len(utxoOpsForBlock) == len(block.Txns)+1 {
+		blockHash, err := block.Hash()
+		if err != nil {
+			// This is bad if this happens.
+			glog.Error(errors.Wrapf(err, "GetTransactionsForConvertBlock: Problem fetching block hash"))
+			return transactions
+		}
+		utxoOpsForBlockLevel := utxoOpsForBlock[len(utxoOpsForBlock)-1]
+		var ops []*types.Operation
+		// Add outputs for Stake Reward Distributions that are not re-staked. We use a fake transaction to
+		// avoid panic in getStakingRewardDistributionOps.
+		implicitOutputOps := node.getImplicitOutputs(&lib.MsgDeSoTxn{TxOutputs: nil}, utxoOpsForBlockLevel, len(ops))
+		ops = append(ops, implicitOutputOps...)
+		// Add outputs for Stake Reward Distributions that are re-staked
+		stakeRewardOps := node.getStakingRewardDistributionOps(utxoOpsForBlockLevel, len(ops))
+		ops = append(ops, stakeRewardOps...)
+
+		transaction := &types.Transaction{
+			TransactionIdentifier: &types.TransactionIdentifier{Hash: fmt.Sprintf("blockHash-%v", blockHash.String())},
+		}
+		transaction.Operations = squashOperations(ops)
+		transactions = append(transactions, transaction)
+	}
+
 	return transactions
 }
```

Comment:
This is a pretty elegant solution. I'd just beef up the comment. Explain how we pay out at the end of an epoch by adding a list of UtxoOperations. Just put all the context here.